### ISSUE TITLE
Bar Graph not ready for negative values

### ISSUE THREAD

**Issue by diversos** [codesuki/react-d3-components#138]
How would you recommend fixing negative values in barChar? I can't explain properly, so I'll show you an example:

![github_eg](https://user-images.githubusercontent.com/14896863/29079716-c88316b0-7c55-11e7-9678-a3bd9c8f94cc.png)

2010: 0+0+0+0
2011: 0+0+0+0
2012: 0+0+0+0
2013: 0+5+8+0
2014: -38+0+13+0
2015: 45+12+130+0
2016: 0+0+0+0

I think that the xAxis should stay in y=0, and the negative values would go under it.

**Comment by codesuki** [comment]
That's interesting! I never considered that.
Let me ask if I understand you correctly.
`13` should be on top of the x-axis and `-38` should be below the x-axis? 
So all negatives stack below and all positives stack above right?

**Comment by diversos** [comment]
Exacly! And several negatives would stack bellow each other. I don't think it should be too hard to implement, since you have the concept at the Scatter Graph.

**Comment by diversos** [comment]
Did you have some time to try it?

**Comment by codesuki** [comment]
I tried it but there are still some unsolved issues.
1) What to do with the labels?
2) Special handling for grouped bar chart

See here:
<img width="512" alt="image" src="https://user-images.githubusercontent.com/5276006/30894172-1bb7f7d0-a37d-11e7-8750-c60fa86ae542.png">

**Comment by codesuki** [comment]
Fixed the grouped chart, but no good idea what to do with the labels.

![image](https://user-images.githubusercontent.com/5276006/30894589-ab487580-a37f-11e7-86fb-272c2e6258ec.png)

**Comment by diversos** [comment]
The only two options I see is either let it in the same place, but over the column bar, or push all the labels down, bellow the graphic.

**Comment by codesuki** [comment]
Yep, I can't see a much better way without total customization. I'll change the rendering order, I hope that's good enough.
